### PR TITLE
ci-op: pipeline imports images from QCI-APPCI

### PR DESCRIPTION
**Pull request by hongkailiu**
This PR handles two cases (probably the only two?) where a CI job imports images from a local cluster.

- the images from a ref on a tag in ci-op's config, such as base_images,
- the images from a release that snapshots the integration streams, i.e., there is `tag_specification` or `releases.integration` as described in https://docs.ci.openshift.org/docs/internals/steps/#releaseimagestagstep

Those images are distributed by the controller `test-images-distributor` in `dptp-controller-manager` https://github.com/openshift/ci-tools/tree/master/pkg/controller/test-images-distributor from app.ci to build-farms.

As planed, we want the Job imports those images from QCI-APPCI, instead of the one cluster that the job is running on.

There is a commit for each of the case above. ~The 3rd commit is about bumping the waiting time on import images which can be done with another PR. It would help the success rate of the job because there are rehearsals (e.g., [job-27m](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/48891/rehearse-48891-pull-ci-openshift-csi-external-attacher-master-e2e-aws-csi/1759423804687257600) and [job2-timeout](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/48891/rehearse-48891-pull-ci-openshift-cluster-nfd-operator-master-e2e-aws/1758686460812201984).
I believe this is because importing from local tags is much faster than quay.io.~

https://issues.redhat.com/browse/DPTP-3878

/cc @openshift/test-platform 

/hold

Let us see if any test breaks.

### CODE REVIEW

**Comment by hongkailiu** [comment]
A good sign from e2e:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_ci-tools/3972/pull-ci-openshift

File: pkg/steps/release/snapshot.go
Comment (hongkailiu) [inline pkg/steps/release/snapshot.go:49]: Before this PR, we import all of the tags including the one that are built in the test and then expect that they are overridden by [OutputImageTagStep](https://docs.ci.openshift.org/docs/internals/steps/#outputimagetagstep).

In the rehearsals, I found out the image points to QCI-APPCI too.
I _believe_ that it is the racing condition on importing from the snapshotting and the tag from OutputImageTagStep.
So here, the imports ignores the built images at snapshotting and wait for OutputImageTagStep to do the job.

**Comment by hongkailiu** [comment]
The failures in e2e seems related.
I did not notice it earlier because the error was not repeating ... G!
I need to fix those.

**Comment by hongkailiu** [comment]
```console
$ oc image mirror registry.build04.ci.openshift.org/ci-op-zqyjxpvw/pipeline@sha256:0bb2b458e5c7da0449a9406da095e5d2195ac2e83beb1697731f499afc5be4cf registry.ci.openshift.org/ci/hongkliu-test:ci-op-ea2ab8a1de26dd211153930819468a5084d5f0b9 -a /tmp/b04.p.json
```

**Review by jmguzik — APPROVED** [review]

File: pkg/steps/release/snapshot.go
Comment (droslean) [inline pkg/steps/release/snapshot.go:90]: Do we care to do this in serial only?

Comment (hongkailiu) [reply pkg/steps/release/snapshot.go:90]: The imports takes places in the OpenShift controllers (I believe they are concurrent).
Here we check if they have done the job before moving forward.
From the tests and rehearsals, the controller update the status on the stream at the same time.
Checking the tags one after another sounds reasonable to me here.
We have other places with a similar checking implementation already.
We might switch to the watch mode in the future if it creates to much workload on the api server by periodic `get`.

Comment (droslean) [reply pkg/steps/release/snapshot.go:90]: Yes, this can be tuned in a follow-up. However, the iteration will wait for the `PollUntilContextTimeout` to stop and then continue. If there is a problem with one image, it will get stuck until timeout either way.

**Comment by hongkailiu** [comment]
/hold

I will do the ICSP first before merging this.

**Pull request merged**
